nfc/src/nfc_platform_zephyr.c

```diff
@@ -14,11 +14,17 @@
 #define LOG_LEVEL CONFIG_NFC_PLATFORM_LOG_LEVEL
 LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
+struct device *clock;
+
 nrfx_err_t nfc_platform_setup(void)
 {
-	nrfx_clock_init(NULL);
-	nrfx_clock_enable();
-	nrfx_clock_hfclk_start();
+	int err;
+
+	clock = device_get_binding(DT_NORDIC_NRF_CLOCK_0_LABEL "_16M");
+	__ASSERT_NO_MSG(clock);
+
+	err = clock_control_on(clock, (void *)1);
```

Comment:
What does this address point to: `(void *)1`?

Reply:
It is strangely done in the clock driver. This pointer is cast to u32_t variable. If it is 1 then CPU is sleep until the 16MHZ clock settles in other case the cpu is not sleep and this is not blocking clock start.

Reply:
Ok then I guess for now we have to use this driver in this weird way, until a proper Clock driver is added in Zephyr.